src/main/scala/is/hail/HailContext.scala

```diff
@@ -534,34 +534,30 @@ class HailContext private(val sc: SparkContext,
     }
   }
 
-  def importMatrix(file: String,
-    fieldHeaders: Option[Array[String]],
-    fieldTypes: Array[Type],
-    optKeyFields: Option[Array[String]],
-    nPartitions: Option[Int] = None,
-    forceBGZ: Boolean = false,
-    dropSamples: Boolean = false,
-    cellType: Type = TInt64(),
-    missingVal: String = "NA",
-    noHeader: Boolean = false): MatrixTable =
-    importMatrices(List(file), fieldHeaders, fieldTypes, optKeyFields, nPartitions, forceBGZ, dropSamples, cellType, missingVal, noHeader)
-
-  def importMatrices(files: List[String],
-    fieldHeaders: Option[Array[String]],
-    fieldTypes: Array[Type],
-    optKeyFields: Option[Array[String]],
-    nPartitions: Option[Int] = None,
-    forceBGZ: Boolean = false,
-    dropSamples: Boolean = false,
-    cellType: Type = TInt64(),
+  def importMatrix(files: java.util.ArrayList[String],
+    rowFields: java.util.HashMap[String, Type],
+    keyNames: java.util.ArrayList[String],
+    cellType: Type,
+    missingVal: String,
+    minPartitions: java.lang.Integer,
+    noHeader: Boolean,
+    forceBGZ: Boolean): MatrixTable =
+    importMatrices(files.asScala, rowFields.asScala.toMap, keyNames.asScala.toArray,
+      cellType, missingVal, if (minPartitions == null) None else Some(minPartitions.toInt), noHeader, forceBGZ)
```

Comment:
We have generally used the `joption` function in python to convert `None` or `T` to `Option[T]`. I think that makes sense here too.